components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/SecondaryUserStoreConfigurator.java

```diff
@@ -144,8 +135,21 @@ private void initializeKeyStore() throws IdentityUserStoreMgtException {
      */
     public String encryptPlainText(String plainText) throws IdentityUserStoreMgtException {
 
+        ServerConfigurationService config =
+                UserStoreConfigComponent.getServerConfigurationService();
+        if (config != null) {
+            String encryptionKeyStore = config.getFirstProperty(ENCRYPTION_KEYSTORE);
+            if (INTERNAL_KEYSTORE.equalsIgnoreCase(encryptionKeyStore)) {
+                try {
+                    return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(plainText.getBytes());
+                } catch (CryptoException e) {
+                    String errorMessage = "Error while encrypting the plain text using internal keystore.";
+                    throw new IdentityUserStoreMgtException(errorMessage, e);
+                }
+            }
+        }
```

Comment:
So how would this switch to symmetric encryption

Reply:
This should be fixed like this right.
- Precedence should be given to the 'UserStorePasswordEncryption' config
- If not present, should switch to default internal encryption mechanism

Reply:
> So how would this switch to symmetric encryption

Switching to symmetric is done on crypto service layer from line 144

Reply:
> This should be fixed like this right.
> 
> * Precedence should be given to the 'UserStorePasswordEncryption' config
> * If not present, should switch to default internal encryption mechanism
Yes the steps should be like that.
Here 'ENCRYPTION_KEYSTORE' constants is the UserStorePasswordEncryption config.